### ISSUE TITLE
Remove steps explaining Qt Linguist from the Translation guidelines

### ISSUE THREAD

**Issue by DelazJ** [qgis/QGIS-Documentation#4101]
In the past, it was only possible to translate the docs, the website and the master branch of QGIS application. If you wanted to update the released versions, you had to pull yourself the sources from github, translate in your own instance of Transifex or using Qt Linguist and then do a pull request. Not straightforward.
Now that we translate all the three active versions of QGIS application (LTR, regular release and dev version) in Transifex, I suggest to remove mention of Qt Linguist in [the translation guidelines](https://docs.qgis.org/testing/en/docs/documentation_guidelines/do_translations.html). Since it's unnecessary, there's no need to confuse contributors...

**Comment by SrNetoChan** [comment]
I disagree with removing. Although transifex makes it very convenient for mass translation, the translation are completely out of context. Some translation teams have people that open their translation in QTLinguist to perform reviews/ clean-ups and small fixes while looking at the dialog they are translating.

We can make it somehow clear that it should be used by advanced users only.

**Comment by DelazJ** [comment]
Close...?
